File: backend/routes/authRoutes.py
from fastapi import APIRouter, HTTPException, status
from schemas.authSchemas import LoginInput, Token
from services import userServices
from services.authServices import create_access_token

router = APIRouter(prefix="/auth", tags=["Auth"])

@router.post("/login", response_model=Token)
def login(payload: LoginInput):
    user_doc = userServices.get_user_doc_by_email(payload.email)
    if not user_doc or not userServices.verify_password(payload.senha, user_doc["senha"]):
        raise HTTPException(status_code=status.HTTP_401_UNAUTHORIZED, detail="Credenciais inválidas")
    token = create_access_token(str(user_doc["_id"]), user_doc["email"])
    return {"access_token": token, "token_type": "bearer"}

File: backend/routes/authRoutes.js

import express from 'express';
import { getUserDocByEmail, verifyPassword } from '../services/userServices.js';
import { createAccessToken } from '../services/authServices.js';

const router = express.Router();

// Login route
router.post('/login', async (req, res) => {
  const { email, senha } = req.body;

  try {
    // Fetch user by email
    const userDoc = await getUserDocByEmail(email);
    if (!userDoc || !verifyPassword(senha, userDoc.senha)) {
      return res.status(401).json({ error: 'Credenciais inválidas' });
    }

    // Generate access token
    const token = createAccessToken(userDoc._id.toString(), userDoc.email);
    res.json({ access_token: token, token_type: 'bearer' });
  } catch (error) {
    res.status(500).json({ error: 'Erro ao processar login' });
  }
});

module.exports = router;